aten/src/ATen/native/README.md

```diff
@@ -0,0 +1,109 @@
+make use of other ATen operations (e.g., it is not necessary to bind into
+TH/THC code).  These functions are declared in this file and then folded
+into the ATen code generation process.
+
+## Registering it in `native_functions.yaml`
+
+The first step is to write an entry for your function in
+`native_functions.yaml`.  The format is as follows:
+
+```
+- func: func_name(ArgType arg0[=default], ArgType arg1[=default], ...) -> ReturnType
+```
+
+ArgType(s) are allowed to be simple types understood by ATen
+(e.g. `Tensor`, `TensorList`, `IntList`, `int64_t`, `double`, ...).
+`Tensor?` means that the tensor is optional; if it is not passed by
+the user it defaults to an undefined tensor (if you declare an argument
+as optional, you must check if it is the case).
+
+ReturnType is allowed to be any ArgType or tuple combination of ArgTypes(s),
+e.g. `(Tensor, Tensor)` defaults are optional and are only allowed to be numbers
+(e.g. '0' for `int64_t`, '5.0' for `double`)
+
+The ATen code generation process will generate a header corresponding
+to the C++ implementation you will have to write.
+The C++ function declarations won't match the declaration here because
+they will undergo the standard ATen C++ transformations, e.g. use of const-ref
+for non-inplace Tensor arguments (instead of `Tensor` you will
+take `const Tensor&`).  So after you finish writing your header, we recommend
+running a build and getting the declaration from `NativeFunctions.h`
+(find it with `find -name NativeFunctions.h`) to copy paste into your
+C++ file.
+
+The declarations also support the following attributes:
+
+```
+variants: function, method
+```
+
+Controls whether Tensor method (`t.foo()`) or namespace Function (`at::foo()`) is
+generated as a result of this declaration.  If the declaration is a method,
+you must have an argument `Tensor self` as the first argument.  In general, you
+should default to defining a new ATen function as `variants: function`, unless
+you know you want something to be usable as a method.
+
+```
+dispatch:
+    CPU: func_cpu
+    CUDA: func_cuda
+```
+
+This specifies the actual name of the function you want to dispatch to, so you
+can dispatch to different functions depending on whether or not you have CPU or
+CUDA tensors.  Technically, it is also possible to write `dispatch: func_name`
+to unconditionally dispatch to a native function whose name is different than
+the name in the public ATen API, but this is generally frowned upon (just name
+them the same thing!)
+
+```
+python_default_init:
+  argument_name: initializing_expression
+```
+
+A map from argument names to default initialize expressions in C++. Such default
+expressions will only be used in Python API. This allows us to write argument
+with a default value that can either cause ambiguity in C++ (e.g., `Scalar p`
+argument in `norm`) or have a type that doesn't allow default value
+None/NULL/nullptr (e.g., `int64_t fft_size` argument in stft, which we want to
```

Comment:
I don't think you should reference explicit examples, those are guaranteed to change and leave you with a dangling reference.  Just make up an example.